source/extensions/compression/gzip/decompressor/config.h

```diff
@@ -0,0 +1,64 @@
+#pragma once
+
+#include "envoy/compression/decompressor/config.h"
+#include "envoy/extensions/compression/gzip/decompressor/v3/gzip.pb.h"
+#include "envoy/extensions/compression/gzip/decompressor/v3/gzip.pb.validate.h"
+
+#include "common/http/headers.h"
+
+#include "extensions/compression/common/decompressor/factory_base.h"
+#include "extensions/compression/gzip/decompressor/zlib_decompressor_impl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Compression {
+namespace Gzip {
+namespace Decompressor {
+
+namespace {
+
+const std::string& gzipStatsPrefix() { CONSTRUCT_ON_FIRST_USE(std::string, "gzip."); }
+const std::string& gzipExtensionName() {
+  CONSTRUCT_ON_FIRST_USE(std::string, "envoy.decompressors.gzip");
+}
+
+} // namespace
+
+class GzipDecompressorFactory : public Envoy::Compression::Decompressor::DecompressorFactory {
+public:
+  GzipDecompressorFactory(const envoy::extensions::compression::gzip::decompressor::v3::Gzip& gzip);
+
+  // Envoy::Compression::Decompressor::DecompressorFactory
```

Comment:
This comment seems to be leaking implementation details abstracted by `createDecompressor()`?

Reply:
Oh this is a convention where we title a section of functions with the name of the abstract class we are implementing, like: https://github.com/envoyproxy/envoy/blob/master/source/server/api_listener_impl.h#L59